src/main/groovy/ai/digital/integration/server/tasks/worker/WorkerOverlaysTask.groovy

```diff
@@ -0,0 +1,55 @@
+package ai.digital.integration.server.tasks.worker
+
+import ai.digital.integration.server.domain.Worker
+import ai.digital.integration.server.util.WorkerUtil
+import org.gradle.api.Action
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Copy
+
+import static ai.digital.integration.server.constant.PluginConstant.PLUGIN_GROUP
+
+class WorkerOverlaysTask extends DefaultTask {
+    static NAME = "workerOverlays"
+
+    static PREFIX = "worker"
+
+    static boolean shouldUnzip(File file) {
+        file.name.endsWith(".zip")
+    }
+
+    WorkerOverlaysTask() {
+        this.configure { ->
+            group = PLUGIN_GROUP
+            mustRunAfter DownloadAndExtractWorkerDistTask.NAME, SyncServerPluginsWithWorkerTask.NAME, SetWorkersLogbackLevelsTask.NAME
+
+            project.afterEvaluate {
+                WorkerUtil.getWorkers(project).each { Worker worker ->
+
+                    if (!worker.overlays.isEmpty() && !WorkerUtil.isExternalRuntimeWorker(worker, project)) {
+                        logger.warn("Overlays on the worker ${worker.name} are ignored because worker's runtime directory is same to the master.")
+                    } else {
+                        worker.overlays.each { Map.Entry<String, List<Object>> overlay ->
+                            def configurationName = "${PREFIX}${overlay.key.capitalize().replace("/", "")}"
```

Comment:
The content of this for each is duplicated already 4 times, let's move it to some shared place.

Reply:
Moved everything that is repeating to the OverlaysUtil.